common-arcana.lic

```diff
@@ -81,14 +81,21 @@ module DRCA
 
   def activate_barb_buff(name)
     ability_data = get_data('spells').barb_abilities[name]
-    case DRC.bput(ability_data['start_command'], ability_data['activated_message'], 'But you are already', 'Your inner fire lacks', 'find yourself lacking the inner fire', 'You should stand', 'You must be sitting')
+    case DRC.bput(ability_data['start_command'], ability_data['activated_message'], 'But you are already', 'Your inner fire lacks', 'find yourself lacking the inner fire', 'You should stand', 'You must be sitting', 'You must be unengaged', 'While swimming?')
+    when 'You must be unengaged'
+      DRC.retreat
+      activate_barb_buff(name)
     when 'You must be sitting'
       DRC.retreat
-      DRC.bput('sit', 'You sit', 'You are already', 'You rise')
+      DRC.bput('sit', 'You sit', 'You are already', 'You rise', 'While swimming?')
```

Comment:
Won't this lead to a loop of sitting and getting that error message repeatedly?

Reply:
I'll do another test and grab the logs to confirm if its entering recursive loop here or just spammy in the combat-trainer event loop trying to activate a buff in an area it'll never be able to.

Maybe worth exploring some kind of backoff/sleep on barb abilities that aren't able to be activated so that they aren't instantly repeated (and fail).

Reply:
Good catch, if you try to sit in too deep of water then now return `false` because won't be able to activate the ability